tests/src/test/java/com/maxio/advancedbilling/controllers/subscriptioncomponents/SubscriptionComponentsDeletePrepaidUsageAllocationTest.java

```diff
@@ -0,0 +1,208 @@
+                        }
+                ),
+                Arguments.arguments(
+                        CreditScheme.REFUND,
+                        (ThrowingConsumer<Integer>) subscriptionId -> {
+                            assertListAllocationsIsEmpty(subscriptionId);
+                            assertThat(getServiceCreditsBalanceInCents(subscriptionId)).isEqualTo(0L);
+
+                            List<CreditNote> creditNotesWithRefunds = getCreditNotesWithRefunds(subscriptionId);
+                            assertThat(creditNotesWithRefunds).hasSize(1);
+
+                            List<InvoiceRefund> refunds = creditNotesWithRefunds.get(0).getRefunds();
+                            assertThat(refunds)
+                                    .hasSize(1)
+                                    .singleElement()
+                                    .satisfies(invoiceRefund -> {
+                                        assertThat(invoiceRefund.getMemo()).isEqualTo("Refund for reverted prepaid usage allocation: 10 units");
+                                        assertThat(invoiceRefund.getOriginalAmount()).isEqualTo("20.0");
+                                        assertThat(invoiceRefund.getAppliedAmount()).isEqualTo("20.0");
+                                    });
+                        }
+                )
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("argsForShouldReturn404ForNotExistentResources")
+    void shouldReturn404ForNotExistentResources(int subscriptionId, int componentId, int allocationId) {
+        // when - then
+        assertNotFound(() -> SUBSCRIPTION_COMPONENTS_CONTROLLER.deletePrepaidUsageAllocation(
+                subscriptionId,
+                componentId,
+                allocationId,
+                new CreditSchemeRequest(CreditScheme.NONE)
+        ));
+    }
+
+    private static Stream<Arguments> argsForShouldReturn404ForNotExistentResources() {
+        return Stream.of(
+                Arguments.arguments(subscription.getId(), prepaidComponent.getId(), 123),
+                Arguments.arguments(subscription.getId(), 123, allocation.getAllocationId()),
+                Arguments.arguments(123, prepaidComponent.getId(), allocation.getAllocationId()),
+                Arguments.arguments(111, 222, 333)
+        );
+    }
+
+    @Test
+    void shouldNotBulkResetWhenProvidingInvalidCredentials() {
+        // when - then
+        assertUnauthorized(() -> TestClient.createInvalidCredentialsClient().getSubscriptionComponentsController()
+                .deletePrepaidUsageAllocation(
+                        subscription.getId(),
+                        prepaidComponent.getId(),
+                        allocation.getAllocationId(),
+                        new CreditSchemeRequest(CreditScheme.NONE)
+                )
+        );
+    }
+
+    private static Component createPrepaidComponentWithAllocationExpiration() throws IOException, ApiException {
```

Comment:
are these settings needed for this particular test? isn't prepaidComponent from testSetup sufficient?

Reply:
There are not, fixed 👍